tasks/managed/populate-release-notes/populate-release-notes.yaml

```diff
@@ -382,6 +382,101 @@ spec:
                     jsonString=$(jq --argjson cves "$CVEsJson" '. += $cves' <<< "$jsonString")
                 fi
 
+                jq --argjson content "$jsonString" '.releaseNotes.content.artifacts += [$content]' "${DATA_FILE}" > \
+                    /tmp/data.tmp && mv /tmp/data.tmp "${DATA_FILE}"
+            done
+        done
+    - name: populate-release-notes-disk-images
+      image: quay.io/konflux-ci/release-service-utils:ea7868ebdcc7a2116c620255034226611d837f42
+      computeResources:
+        limits:
+          memory: 32Mi
+        requests:
+          memory: 32Mi
+          cpu: 10m
+      script: |
+        #!/usr/bin/env bash
+        set -ex
+
+        DATA_FILE="$(params.dataDir)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No data JSON was provided."
+            exit 1
+        fi
+
+        # Try to extract the first contentType if it exists
+        content_type=$(jq -r '.mapping.components[]?.contentGateway?.contentType // empty' "$DATA_FILE" | head -n1)
+
+        if [ "$content_type" != "disk-image" ]; then
+            echo "Not disk-image content. Skipping disk-image-specific logic."
+            exit 0
+        fi
+
+        SNAPSHOT_FILE="$(params.dataDir)/$(params.snapshotPath)"
+        if [ ! -f "${SNAPSHOT_FILE}" ] ; then
+            echo "No valid snapshot file was provided."
+            exit 1
+        fi
+
+        NUM_COMPONENTS=$(jq '.components | length' "${SNAPSHOT_FILE}")
+        for ((i = 0; i < NUM_COMPONENTS; i++))
+        do
+            component=$(jq -c --argjson i "$i" '.components[$i]' "${SNAPSHOT_FILE}")
+            name=$(jq -r '.name' <<< "$component")
+
+            # Construct CVE json
+            CVEsJson='{"cves":{"fixed":{}}}'
+            CVES=$(jq -c '[.releaseNotes.cves[]? | select(.component=="'"$name"'")]' "${DATA_FILE}")
+            NUM_CVES=$(jq 'length' <<< "$CVES")
+            for ((j = 0; j < NUM_CVES; j++)); do
+                cve=$(jq -c --argjson j "$j" '.[$j]' <<< "$CVES")
+                cveJson=$(jq -n \
+                    --arg id "$(jq -r '.key' <<< "$cve")" \
+                    --argjson packages "$(jq -c '.packages // []' <<< "$cve")" \
+                    '{($id): {"packages": $packages}}')
+                CVEsJson=$(jq --argjson cve "$cveJson" '.cves.fixed += $cve' <<< "$CVEsJson")
+            done
+
+            # Get the number of staged files for this component (disk-images use staged.files structure)
+            FILES_LENGTH=$(jq --arg name "$name" \
+              '[.mapping.components[]
+                | select(.name == $name)
+                | .staged.files
+                | length][0]' "$DATA_FILE")
+
+            for ((k = 0; k < FILES_LENGTH; k++)); do
+                file=$(jq -c --arg name "$name" --argjson k "$k" \
+                    '.mapping.components[]
+                      | select(.name == $name)
+                      | .staged.files[$k]' "$DATA_FILE")
```

Comment:
this through line 463 look to be the only differences between this step and the binary one. Can't we just combine into one step and just set arch and os appropriately based on the contentType?